torch/_meta_registrations.py

```diff
@@ -135,3 +125,46 @@ def meta_adaptive_avg_pool2d(self, output_size):
 def meta_adaptive_avg_pool3d(self, output_size):
     check(self.ndim == 4 or self.ndim == 5, f"Expected 4D or 5D tensor, but got {self.shape}")
     return self.new_empty(self.shape[:-3] + tuple(output_size))
+
+@torch.library.impl(meta_lib, "index.Tensor")
+def meta_index_Tensor(self, indices):
+    # aten::index is the internal advanced indexing implementation
+    check(len(indices) <= self.ndim, f"too many indices for tensor of dimension {self.ndim} (got {len(indices)})")
+    # checkIndexTensorTypes
+    result = []
+    for i, index in enumerate(indices):
+        if index is not None:
+            check(index.dtype in [torch.long, torch.int8, torch.bool], "tensors used as indices must be long, byte or bool tensors")
+            if index.dtype in [torch.int8, torch.bool]:
+                nonzero = index.nonzero()
+                for j in range(index.ndim):
+                    result.append(nonzero.select(1, j))
+            else:
+                result.append(index)
+        else:
+            result.append(index)
+    indices = result
+    # expand_outplace
+    indices = list(refs._maybe_broadcast(*indices))
+    # add missing null tensors
+    while len(indices) < self.ndim:
+        indices.append(None)
+    # AdvancedIndex::AdvancedIndex
+    # "In general, the shape of the resultant array will be the concatenation of
+    # the shape of the index array (or the shape that all the index arrays were
+    # broadcast to) with the shape of any unused dimensions (those not indexed)
+    # in the array being indexed."
+    #
+    # But Nones at the beginning of the list get treated specially
```

Comment:
I don't see where it's doing an analog of transposeToFront https://github.com/pytorch/pytorch/blob/96deba836ad80ccf0c95bbc25f699de2bf367430/aten/src/ATen/native/TensorAdvancedIndexing.cpp#L508-L512 - is it not necessary?

Reply:
What might be happening is that I am not doing strides correctly. I originally had thought tests passing put me in the clear but in fact strides are disabled lol

Reply:
It should be fine then, all unindexed dimensions after there has been some indexed dimension are crowded out to the back, which is what should happen.

Reply:
Ah, no, I think you'd get this one wrong:
```
In [6]: a=torch.randn(3,4,5,6)

In [7]: a[:, [0,1],:,[0,1]].shape
Out[7]: torch.Size([2, 3, 5])
```
You'd have `3` as before_shape, whereas it should be after_shape?

Reply:
yup you're right. adding this to the opinfos

Reply:
@ngimel ok pushed the fix